### ISSUE TITLE
Add linear API

### ISSUE THREAD

**Issue by tmigot** [JuliaSmoothOptimizers/AmplNLReader.jl#127]
See NLPModels.jl 0.18/0.19

I have a draft here https://github.com/JuliaSmoothOptimizers/AmplNLReader.jl/tree/add-linear-API but we probably should add functions to the C interface.

**Comment by amontoison** [comment]
@tmigot Do you think that we can use your branch `add-linear-API` to fix the CI tests?

**Comment by tmigot** [comment]
Yes that's a good idea. I think we were waiting for somebody to grab this.

Do you need me to open the PR?

**Comment by amontoison** [comment]
Yes, I can continue it if you explain me what is missing.
Because the PR is quite old, I'm a little bit lost.

**Comment by tmigot** [comment]
No problem, I opened the PR and tried to add some comments. My PR was not ideal as it was only using the functions from Ampl API that were already exported. It might be that they have some tailored functions that we didn't have.
